src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -160,5 +162,61 @@ internal static int RouteComparison(RouteEntry x, RouteEntry y)
 ");
             }
         }
+
+        private readonly struct Key : IEquatable<Key>
+        {
+            public readonly Assembly[] Assemblies;
+
+            public Key(Assembly[] assemblies)
+            {
+                Assemblies = assemblies;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is Key other ? base.Equals(other) : false;
+            }
+
+            public bool Equals(Key other)
+            {
+                if (Assemblies == null && other.Assemblies == null)
+                {
+                    return true;
+                }
+                else if (Assemblies == null ^ other.Assemblies == null)
+                {
+                    return false;
+                }
+                else if (Assemblies.Length != other.Assemblies.Length)
+                {
+                    return false;
+                }
+
+                for (var i = 0; i < Assemblies.Length; i++)
+                {
+                    if (!Assemblies[i].Equals(other.Assemblies[i]))
```

Comment:
Are these always in a stable order?

Reply:
Does that matter?

Reply:
Yes. They are ordered by `RouteTableFactory.Create`